internal/osutils/autostart/autostart_lin.go

```diff
@@ -58,6 +63,18 @@ func (a *App) Disable() error {
 }
 
 func (a *App) IsEnabled() (bool, error) {
+	appDir, err := dirPath(applicationDir)
+	if err != nil {
+		return false, errs.Wrap(err, "Could not get application file")
+	}
+	appPath := filepath.Join(appDir, launchFile)
+
+	if !fileutils.FileExists(appPath) {
+		if err = setupAppFile(appPath); err != nil {
+			return false, errs.Wrap(err, "Could not setup app file")
+		}
+	}
```

Comment:
`IsEnabled()` is for checking if it's enabled.... that's it. This function should not be creating anything.

Reply:
The type that is holding the context (the method receiver `*App`) does not have a part of its surface that makes setting up the desktop file make sense. I thought that the desktop file should be added in the installation process, but I was not sure since the AC of my story includes it. Can I make a story to later move the logic into the installer?

Reply:
`Enable()` seems like a pretty sensible place to me? Much more sensible than `IsEnabled()`.

Reply:
IsEnabled is run more often in the control flow rather than strictly when autostart is enabled manually. Using "Enable" for this logic means the application will not show up in the launcher until the application is enabled.

The desktop shortcut file creation should probably be an installation step, and not even in this code. Given the existing code and AC, I still suggest using IsEnabled along with a story to organize/model these things better.

Reply:
Oh I see what you're saying. You're mixing the launcher into the autostart behaviour. For the launcher a more appropriate place would be:

https://github.com/ActiveState/cli/blob/master/internal/runners/prepare/prepare_windows.go#L40-L40

I thought I already had a followup story to remove `state _prepare` in favour of the new installer, but can't seem to find it. Logged a new one:

https://www.pivotaltracker.com/story/show/177913672